Geometry/GEMGeometryBuilder/src/GEMGeometryBuilderFromDDD.cc

```diff
@@ -263,3 +294,208 @@ GEMGeometryBuilderFromDDD::RCPBoundPlane GEMGeometryBuilderFromDDD::boundPlane(c
 
   return RCPBoundPlane(new BoundPlane(posResult, rotResult, bounds));
 }
+
+// DD4HEP
+
+void GEMGeometryBuilderFromDDD::build(GEMGeometry& theGeometry,
+                                      const cms::DDCompactView* cview,
+                                      const cms::MuonNumbering& muonConstants) {
+  std::string attribute = "MuStructure";
+  std::string value = "MuonEndCapGEM";
+  cms::DDFilteredView fv(cview->detector(), cview->detector()->worldVolume());
+  cms::DDSpecParRefs refs;
+  const cms::DDSpecParRegistry& mypar = cview->specpars();
+  mypar.filter(refs, attribute, value);
+  fv.mergedSpecifics(refs);
+
+  bool doChambers = fv.firstChild();
+
+  // loop over superchambers
+  std::vector<GEMSuperChamber*> superChambers;
+
+  while (doChambers) {
+    MuonBaseNumber mbn = muonConstants.geoHistoryToBaseNumber(fv.history());
+    cms::GEMNumberingScheme gemnum(muonConstants.values());
+    gemnum.baseNumberToUnitNumber(mbn);
+    GEMDetId detIdCh = GEMDetId(gemnum.getDetId());
+
+    if (detIdCh.layer() == 1) {  // only make superChambers when doing layer 1
+      GEMSuperChamber* gemSuperChamber = buildSuperChamber(fv, detIdCh);
+      superChambers.push_back(gemSuperChamber);
+    }
+
+    GEMChamber* gemChamber = buildChamber(fv, detIdCh);
+
+    fv.down();
+    fv.down();
+    bool doEtaPart = fv.nextSibling();
+    while (doEtaPart) {
+      MuonBaseNumber mbn = muonConstants.geoHistoryToBaseNumber(fv.history());
+      cms::GEMNumberingScheme gemnum(muonConstants.values());
+      gemnum.baseNumberToUnitNumber(mbn);
+      GEMDetId detId = GEMDetId(gemnum.getDetId());
+      GEMEtaPartition* etaPart = buildEtaPartition(fv, detId);
+
+      gemChamber->add(etaPart);
+
+      theGeometry.add(etaPart);
+      doEtaPart = fv.sibling();
+    }
+    theGeometry.add(gemChamber);
+    fv.up();
+    fv.up();
+    fv.up();
+
+    doChambers = fv.firstChild();
+
+  }  // close while Chambers
+
+  // construct the regions, stations and rings.
+  for (int re = -1; re <= 1; re = re + 2) {
+    GEMRegion* region = new GEMRegion(re);
```

Comment:
Beyond the scope of this PR, but all the `new `s here make me very nervous. It would be better to use unique_ptrs and then `move()` them into their owners. It would be even better for their owners to create and manage them immediately, with accessors for the created objects.